tests/tests_app/core/test_lightning_flow.py

```diff
@@ -604,6 +605,7 @@ def run(self):
                 # TODO (tchaton) Optimize flow execution.
                 assert 4.0 < abs(time() - self._last_time) < 6.0
                 self._exit()
+        self.counter += 1
```

Comment:
I'm not sure why this is needed. The `SchedulerThread` is not reliant on the event loop iterating, so this should happen without state being forced to update.

Can you clarify @tchaton?

Reply:
There were a bug in `_delta_to_app_state_delta`. Removed the counter.